### PR TITLE
chore: Refactor journey tests, remove sauce tunnel requirement

### PR DESCRIPTION
**Pull request by bzang**
This is a refactor of journey tests to use page objects as a way to simplify our tests, remove flakiness, and make updates much easier. This PR also has the following:
- Netlify deployments
- Add package-lock
- Updated dependencies that don't break anything
- split wdio configs into separate files
- smaller files for file sharing tests

@adamweeks You've reviewed a decent bit of this already in the previous PR. You'll want to review:
- https://github.com/ciscospark/react-ciscospark/commit/1c295d4d65c57fd9ea8b5c9cdfc162fd35a31cf5
- https://github.com/ciscospark/react-ciscospark/commit/2375698e46722b2a76c46ef56ea075d9533d0a3b

This PR supersedes #700.

### CODE REVIEW

File: packages/node_modules/@ciscospark/react-component-chip-base/src/index.js [hunk 1]
```diff
@@ -31,7 +31,11 @@ function ChipBase(props) {
     <div className={classNames('ciscospark-chip', styles.chip)}>
       {children}
       <div className={classNames('ciscospark-chip-action', styles.action)}>
-        <Button iconType={ICON_TYPE_DELETE} onClick={handleRemove} />
+        <Button
+          iconType={ICON_TYPE_DELETE}
+          onClick={handleRemove}
+          accessibilityLabel="Delete Share"
```
Comment (bzang) [inline packages/node_modules/@ciscospark/react-component-chip-base/src/index.js]: Needed so tests can select this button.

**Comment by adamweeks** [comment]
Deploy preview for *hopeful-hamilton-4090b9* ready!

Built with commit bcd7045d3d93eb0c420ca403beb07a5045612ee5

https://deploy-preview-708--hopeful-hamilton-4090b9.netlify.com

File: test/journeys/.eslintrc.js [hunk 2]
```diff
@@ -8,6 +8,11 @@ module.exports = {
     $: true
   },
   rules: {
-    'no-console': 'off'
+    'no-console': 'off',
+    'prefer-arrow-callback': [
+      'error', {
+        allowNamedFunctions: true
```
Comment (bzang) [inline test/journeys/.eslintrc.js]: Needed to allow named mocha functions for `this.retries()`

Comment (adamweeks) [reply test/journeys/.eslintrc.js]: Good to know!

File: test/journeys/lib/widgetPages/base.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  removeWidget() {
+    this.browser.execute(() => {
+      window.removeWidget();
+    });
+  }
+
+  setPageTestName(testName) {
+    console.log(testName);
```
Comment (adamweeks) [inline test/journeys/lib/widgetPages/base.js]: Log needed?

File: test/journeys/lib/widgetPages/base.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+import uuid from 'uuid';
+
+export default class BaseWidgetObject {
+  get hasWidgetContainer() { return this.browser.isVisible(this.elements.widgetContainer); }
+
+  get hasWidget() { return this.browser.isVisible(this.elements.widgetBody); }
+
+  constructor({aBrowser, user}) {
```
Comment (adamweeks) [inline test/journeys/lib/widgetPages/base.js]: Some jsdoc would be good here. Defining what all is in the user object would be helpful.

File: test/journeys/lib/widgetPages/space/messaging.js [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+  }
+
+  get lastSuccessfulActivityText() { return this.browser.getText(this.elements.lastSuccessfulActivityText); }
+
+  get lastFileSize() { return this.browser.getText(this.elements.fileSize); }
+
+  constructor(props) {
+    super(props);
+
+    const activityContainers = '//div[contains(@class, "ciscospark-activity-item-container")]';
```
Comment (adamweeks) [inline test/journeys/lib/widgetPages/space/messaging.js]: Why are some of these defined up here and others within the elements object?

Comment (bzang) [reply test/journeys/lib/widgetPages/space/messaging.js]: If it's used to compose a selector in the `elements` object, we have to define it first before we can use it. I tried to keep anything that isn't used in a test or getter out of the actual `elements` object

File: test/journeys/lib/widgetPages/base.js (hunk 4 above)
Comment (bzang) [reply test/journeys/lib/widgetPages/base.js]: Good call.

**Pull request closed without merging**